src/mds/MDCache.cc

```diff
@@ -10276,28 +10308,30 @@ CDir *MDCache::add_replica_dir(bufferlist::iterator& p, CInode *diri, mds_rank_t
 	      << diri->dirfragtree << dendl;
       diri->dirfragtree.force_to_leaf(g_ceph_context, df.frag);
     }
-
     // add replica.
     dir = diri->add_dirfrag( new CDir(diri, df.frag, this, false) );
-    dir->decode_replica(p);
-
+    __u32 nonce;
+    decode(nonce, p);
+    dir->set_replica_nonce(nonce);
+    dir->_decode_base(p);
     // is this a dir_auth delegation boundary?
     if (from != diri->authority().first ||
 	diri->is_ambiguous_auth() ||
 	diri->is_base())
       adjust_subtree_auth(dir, from);
     
     dout(7) << "add_replica_dir added " << *dir << " nonce " << dir->replica_nonce << dendl;
-    
     // get waiters
     diri->take_dir_waiting(df.frag, finished);
   }
+  DECODE_FINISH(p);
 
   return dir;
 }
 
 CDentry *MDCache::add_replica_dentry(bufferlist::iterator& p, CDir *dir, list<MDSInternalContextBase*>& finished)
 {
+  DECODE_START_LEGACY_COMPAT_LEN(1, 1, 1, p);
```

Comment:
using DECODE_START. we don't care compatibility for now

Reply:
oh, thanks, will change them

Reply:
updated